ensime_shared/ensime.py

```diff
@@ -136,16 +136,23 @@ def queue_poll(self, sleep_t=0.5):
 
         Value of sleep is low to improve responsiveness.
         """
+        connection_alive = True
         while self.running:
             if self.ws:
                 def logger_and_close(m):
                     self.log("Websocket exception: {}".format(m))
-                    self.teardown()
+                    if not self.running:
```

Comment:
Killing the server is too extrem here because I think that we're checking if the server is alive and when it's not we explicitly call `teardown` as you see in the logs `teardown: in`. The statement we're discussing is covering the case in which teardown is not called and that may happen because of other spurious errors. My take is that the ws thread and the plugin are independent and, therefore, the ws thread may catch an exception when the vim plugin hasn't exited. Does that make sense?